### ISSUE TITLE
integrate evaluation with checkpointing schedule 

### ISSUE THREAD

**Issue by RobotSail** [Red-Hat-AI-Innovation-Team/training_hub#28]
We support training schedules of different variations such as epoch-based, step-based, token-based, etc.

Since we expose the ability to run validation loss during our training loop, we should also be able to use the same mechanism for computing validation.
